operator/pkg/model/translation/gateway-api/translator.go

```diff
@@ -110,7 +111,15 @@ func (t *gatewayAPITranslator) Translate(m *model.Model) (*ciliumv2.CiliumEnvoyC
 		lbSvc.Spec.ExternalTrafficPolicy = corev1.ServiceExternalTrafficPolicy("")
 	}
 
-	return cec, lbSvc, getEndpoints(*source), err
+	shortedName := model.Shorten(source.Name)
+	cec.Labels = mergeMap(cec.Labels, mergeMap(allLabels, map[string]string{
+		gatewayNameLabel: shortedName,
+	}))
+	if len(allAnnotations) > 0 {
+		cec.Annotations = mergeMap(cec.Annotations, allAnnotations)
+	}
+
+	return cec, lbSvc, getEndpoints(*source, allLabels, allAnnotations), err
```

Comment:
for the sake of readability and treating it the same as the Service i would assign `getEndpoints(*source, allLabels, allAnnotations)` to a variable.